python/ray/serve/api.py

```diff
@@ -638,7 +642,70 @@ async def __call__(self, request):
         number if you want to save memory on the node resource.
     """
 
-    raise NotImplementedError("Multiplexed deployment is not supported yet.")
+    if func is not None:
+        if not callable(func):
+            raise TypeError(
+                "The `multiplexed` decorator must be used with a function or method."
+            )
+
+        if not inspect.iscoroutinefunction(func):
+            raise TypeError(
+                "@serve.multiplexed can only be used to decorate async "
+                "functions or methods."
+            )
+        signature = inspect.signature(func)
+        if len(signature.parameters) == 0 or len(signature.parameters) > 2:
+            raise TypeError(
+                "@serve.multiplexed can only be used to decorate functions or methods "
+                "with at least one 'model_id: str' argument."
+            )
```

Comment:
I think this should enforce exactly one argument (the model ID) as per our discussion and the docstring?

Reply:
User can have `func(model_id: str)` or `func(self, model:str)`.

Reply:
ohhhhh.... got it